File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {        
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Integer> queue = new LinkedList<>();

        for(int i = 0; i < priorities.length; i++){
            queue.add(i);
        }

        while(!queue.isEmpty()){
            int previousSize = queue.size();
            int top = queue.poll();
            
            for (int idx : queue){
                if (priorities[idx] > priorities[top]){
                    queue.add(top);
                    break;
                }
            }
            
            if (queue.size() != previousSize){
                answer++;
                if (top == location) break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    
    queue = deque(priorities)
    while queue:
        m = max(queue)
        l = queue.popleft()
        #변수 l에 저장하면서 popleft()했으므로 타겟 프로세스의 위치가 1 줄어들었다. 고로, location--
        location-=1
        
        if l!=m:
            queue.append(l)
            if location < 0:
                location = len(queue)-1
        else:
            answer+=1
            print("현재 값:", l, answer, location)
            if location<0:
                break
        print(queue)
    return answer